src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -755,19 +755,7 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
         private static List<PropertyInfo> GetAllProperties(
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)]
             Type type)
-        {
-            var allProperties = new List<PropertyInfo>();
-
-            Type? baseType = type;
-            do
-            {
-                allProperties.AddRange(baseType!.GetProperties(DeclaredOnlyLookup));
-                baseType = baseType.BaseType;
-            }
-            while (baseType != typeof(object));
-
-            return allProperties;
-        }
+        => type.GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static).ToList();
```

Comment:
```suggestion
        => type.GetProperties(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.Static);
```

There's no reason to copy this array into a list - it is just extra allocation and copying around data unnecessarily.